src/main/xerus/monstercat/api/APIConnection.kt

```diff
@@ -200,11 +215,34 @@ class APIConnection(vararg path: String) : HTTPQuery<APIConnection>() {
 			return ConnectResult(connectsid, validity, session)
 		}
 		
+		fun login(username: String, password: String): Boolean {
+			val connection = APIConnection("v2", "signin")
+			val context = HttpClientContext()
+			connection.post(HttpPost(connection.uri).apply {
+				setHeader("Accept", "application/json")
+				setHeader("Content-type", "application/json")
+				entity = StringEntity("""{"email":"$username","password":"$password"}""")
+			}, context)
+
+			val code = connection.response?.statusLine?.statusCode ?: 0
```

Comment:
why the "?: 0" ?

Reply:
but why not leave it as `null`, to make it clear there was no value?

Reply:
it's an int ? and we already do a check on whether the response code is on the 200-OK range, so 0 never passes

Reply:
Removed it. It is an int, so if it is null, it isn't in the 200 range. 94d29ef1915da2c73fe8451d994bf5e47c56b101